source/features/releases-tab.tsx

```diff
@@ -69,18 +69,15 @@ async function addReleasesTab(): Promise<false | void> {
 				<span data-content="Releases">Releases</span>
 				{count && <span className="Counter" title={count > 999 ? String(count) : ''}>{abbreviateNumber(count)}</span>}
 			</a>
-		</li>
+		</li>,
 	);
-	repoNavigationBar.append(releasesTab);
-
-	// This re-triggers the overflow listener forcing it to also hide this tab if necessary #3347
-	repoNavigationBar.replaceWith(repoNavigationBar);
```

Comment:
This doesn't seem to have any effect anymore (if it ever had).

Reply:
I'm guessing the window `resize` event, but as for the initial page load, I haven't seen any custom events being fired so I don't really know.

What I don't get is why only the "Releases" tab is affected by this — and only when navigating with pjax.

Reply:
This line was supposed to trigger the component to update the list of children to toggle on the next resize, so `resize` isn't it.

What happens is, if I remember correctly:

1. page loads 10 items in nav bar
1. GitHub detects 10 items
1. RGH adds 2 items
1. GitHub will not track these 2 new items

steps 2 and 3 might happen in the alternate order and GitHub might handle all 12 items, but it's basically a race condition. This line was supposed to avoid that.

Reply:
> it's basically a race condition

Yeah, replacing the element with itself actually works… as long as it's done late enough. Since afaik there's no event we could listen to, the only fix I can think of is

```ts
await delay(50);
```

Reply:
This may not be a race condition, but just a GitHub bug/crappy implementation.

The reflow of the tabs actually happens *seconds* after the page is loaded — if anything, we make it less obvious by hiding some tabs very early during page load.

Since afaik there's no event we could listen to, unless we add a three-second delay in the feature code, any DOM trick we try to pull off is going to happen way too soon and not affect anything.

Possible solution: add a timeout of ~10 seconds (maybe less), and at the end of it check if the "Release" tab is still visible and overlapping with the dropdown (cf. video above). If yes, hide it and bring the dropdown item back.